docs/maybe.md

```diff
@@ -0,0 +1,52 @@
+###E247.Fun
+#Maybe&lt;T&gt;
+
+```Maybe<T>``` is used to represent a value that might not be available. While it is an option to use ```null``` for reference types, it isn't explicit in the type system and returning ```null``` can lead to problems when it's not handled properly, using ```Maybe<T>``` instead makes it obvious that the potential lack of a value needs to be handled, and provides some handy methods for dealing with boilerplate based around that.
```

Comment:
Instead of "it isn't explicit in the type system", I'd say "it isn't explicit in the type".